lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -173,6 +175,11 @@ def initialize(config_root, config_path, loaded_config)
     @data_providers = nil
   end
 
+  def fail(issue, args = EMPTY_HASH, line = nil)
```

Comment:
Was it intentional to override/redefine ruby's `fail` here with our own (`fail` originaly being an alias to `raise`)? That seems like it could surprise someone.